modules/registrars/realtimeregister/src/Assets/Tpl/admin/widget/error_log.tpl

```diff
@@ -1,97 +1,192 @@
+            method: 'POST',
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/x-www-form-urlencoded',
+            },
+            body: new URLSearchParams({
+                action: 'fetchErrorLogEntries',
+                module: 'realtimeregister',
+                pageId: pageId,
+                searchTerm: query
+            })
+        }).then(async(asyncResponse) => {
+            const response = await asyncResponse.json();
+            if (response.result === 'success') {
+                logs.clear();
+                resultLogPage.innerHTML = '';
 
-    <style>
-        .log-container {
-            height: 500px;
-            overflow-y: scroll;
+                response.logEntries.forEach(log => {
+                    let elm = document.createElement('div');
+                    elm.className = 'log';
 
-            &:first-child {
-                margin-top: 1rem;
+                    let smallElm = document.createElement('span');
+                    smallElm.className = 'log-created';
+                    smallElm.innerText = log.created_at;
+                    elm.appendChild(smallElm);
+
+                    let buttonElm = document.createElement('button');
+                    buttonElm.classList.add(...['log-details', 'btn', 'btn-default', 'block']);
+                    buttonElm.innerHTML = '<small data-log-id="' + log.id + '">{$LANG.rtr.widgets.errorlog.show_detail}</small>';
+                    buttonElm.setAttribute('data-log-id', log.id);
+                    buttonElm.addEventListener('click', onLogClick);
+                    elm.appendChild(buttonElm);
+
+                    let logMessageElm = document.createElement('span');
+                    logMessageElm.classList.add('log-message');
+                    logMessageElm.innerHTML = log.message;
+                    elm.appendChild(logMessageElm);
+
+                    elm.appendChild(document.createElement('hr'));
+
+                    resultLogPage.appendChild(elm);
+
+                    logs.set(log.id, log);
+                });
+
+                document.getElementById('previous-log-page').classList.add('hidden');
+                document.getElementById('next-log-page').classList.add('hidden');
+                if (response.pageId > 1) {
+                   document.getElementById('previous-log-page').classList.remove('hidden');
+                }
+                if (response.hasMorePages === true) {
+                   document.getElementById('next-log-page').classList.remove('hidden');
+                }
+                document.getElementById('results-log-waiting-for-input').classList.add('hidden');
+
+                if (pageId === 1 && response.logEntries.length === 0) {
+                    document.getElementById('results-log-empty').className.remove('hidden');
+                }
             }
+            document.getElementById('log-search-form').disabled = false;
```

Comment:
why use so much default javascript instead of jquery? Is not consistent with the rest of the module

Reply:
jQuery was cool when it was created ( 2006), because a lot of browsers had their own interpretation about how javascript should work. Since that time a lot things changes (eg. ECMAscript),so I don't see a reason to keep using it.

Reply:
fair enough